### ISSUE TITLE
Cannot instantiate Poisson model

### ISSUE THREAD

**Issue by jmxpearson** [blei-lab/edward#322]
Thanks for a wonderful package!

The following

```python
import edward
ed.models.Poisson(lam=tf.ones(2))
```
fails with
```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-89-6c955130691d> in <module>()
----> 1 ed.models.Poisson(lam=tf.ones(2))

/Users/jmxp/anaconda/lib/python3.5/site-packages/edward/models/random_variables.py in __init__(self, *args, **kwargs)
     33     class _WrapperRandomVariable(RandomVariable, _candidate):
     34       def __init__(self, *args, **kwargs):
---> 35         RandomVariable.__init__(self, *args, **kwargs)
     36 
     37     _WrapperRandomVariable.__name__ = _name

/Users/jmxp/anaconda/lib/python3.5/site-packages/edward/models/random_variable.py in __init__(self, *args, **kwargs)
     58     super(RandomVariable, self).__init__(*args, **kwargs)
     59     tf.add_to_collection(RANDOM_VARIABLE_COLLECTION, self)
---> 60     self._value = self.sample()
     61 
     62   def __str__(self):

/Users/jmxp/anaconda/lib/python3.5/site-packages/tensorflow/contrib/distributions/python/ops/distribution.py in sample(self, sample_shape, seed, name)
    372         return self.sample_n(sample_shape, seed)
    373       sample_shape, total = self._expand_sample_shape(sample_shape)
--> 374       samples = self.sample_n(total, seed)
    375       output_shape = array_ops.concat(0, [sample_shape, array_ops.slice(
    376           array_ops.shape(samples), [1], [-1])])

/Users/jmxp/anaconda/lib/python3.5/site-packages/tensorflow/contrib/distributions/python/ops/distribution.py in sample_n(self, n, seed, name)
    395       TypeError: if `n` is not an integer type.
    396     """
--> 397     self._check_hasattr(self._sample_n)
    398     with self._name_scope(name, values=[n]):
    399       n = ops.convert_to_tensor(n, name="n")

AttributeError: 'Poisson' object has no attribute '_sample_n'
```

It would be nice if [this line](https://github.com/blei-lab/edward/blob/b4c72e29deb94170138164ac011a886edbc11457/edward/models/random_variable.py#L60) were modified to at least allow the user to specify `self._value` via keyword argument, since TensorFlow still lacks a number of samplers.

**Comment by dustinvtran** [comment]
hey @jmxpearson, thanks for checking it out. 

i think that proposed solution makes sense. we should perhaps also include type and shape checking to see that `self._value` has the same type as the underlying distribution, and matches the shape `batch_shape + event_shape` ([source](https://www.tensorflow.org/versions/r0.11/api_docs/python/contrib.distributions.html)).

**Comment by jmxpearson** [comment]
Sounds good. Do you prefer `value` or perhaps `init` as the keyword?

**Comment by dustinvtran** [comment]
keeping with pep8, i'd say `value`.

**Comment by jmxpearson** [comment]
cf #326

**Comment by dustinvtran** [comment]
fixed via #326
